v3/widgets/barchart.go

```diff
@@ -87,3 +87,21 @@ func (self *BarChart) Draw(buf *Buffer) {
 		barXCoordinate += (self.BarWidth + self.BarGap)
 	}
 }
+
+//
+// Compute bar text position based on character length.
+//
+func (self *BarChart) calcNumberXPos(data float64) int {
+	numFormatterData := self.NumFormatter(data)
```

Comment:
suggestion: since this function is unexported, it could be static instead of instance method. using self.NumFormatter before call the function and removing the instance aggregation (self *BarChart)

```go
line 74:  numberXCoordinate := barXCoordinate + calcNumberXPos(self.NumFormatter(data))
//
// Compute bar text position based on character length.
//
func calcNumberXPos(data float64) int {
	numberCharsLen := len(numFormatterData)
	barWidthCenter := int(float64(self.BarWidth / 2))

	var numberXCharPos int = barWidthCenter - numberCharsLen + (numberCharsLen / 2)

	if numberCharsLen > barWidthCenter {
		numberXCharPos = numberCharsLen - barWidthCenter
	}

	return int(numberXCharPos)
}
```

Reply:
@fafg While I have converted `calcNumberXPos` to a static method I just realized the `termui` package structure has changed since this PR was created.  Fixing this now.